compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -519,6 +519,22 @@ impl<'hir> Map<'hir> {
         }
     }
 
+    /// Walks the contents of a crate. See also `Crate::visit_all_items`.
+    pub fn walk_crate(self, visitor: &mut impl Visitor<'hir>) {
```

Comment:
Could you rename this to something like `walk_crate_without_crate_attrs`, to make it clear that it doesn't walk everything in the crate?